includes/wikia/models/DesignSystemCommunityHeaderModel.class.php

```diff
@@ -475,4 +500,52 @@ private function formatLocalNavData( array $items, int $nestingLevel ): array {
 	private function getSpecialPageURL( $name ): string {
 		return SpecialPage::getTitleFor( $name )->getLocalURL();
 	}
+
+	private function getFandomStoreDataFromCache() {
+		global $wgMemc, $wgCityId;
+
+		$memcKey = wfMemcKey( DesignSystemApiController::MEMC_PREFIX_FANDOM_STORE, $wgCityId );
+		$slicePos = strpos( $memcKey, ':' );
+		$memcKey = substr( $memcKey, $slicePos + 1 );
+		$cachedStoreData = $wgMemc->get( $memcKey );
+
+		return !empty( $cachedStoreData->results ) ? $this->formatFandomStoreData( $cachedStoreData->results ) : null;
+	}
+
+	private function formatFandomStoreData( $apiData ) {
+		// get copy of links from api data
+		$arrObj = new ArrayObject( $apiData );
+		$links = $arrObj->getArrayCopy();
+
+		// remove store item object from list
+		$firstItem = array_shift( $links );
+
+		// Only display 9 items + show more link
+		if ( count( $apiData ) > self::FANDOM_STORE_ITEM_LIMIT ) {
```

Comment:
Let's add a check to make sure that `$firstItem` actually has a `url` property on it. Along with any other property that might not exist. If something breaks here we can have a fatal error in the navigation causing the page to break. 

We can return `null` from this function and nothing will render.

Reply:
This actually won't break, because if there's no url, it falls back to an empty string in the code.

`<a href="<?= Sanitizer::encodeAttribute( $exploreItem->href ) ?? '#' ?>"`

Reply:
Added error handling 645e587.